### DISCUSSION TITLE
Domoticz MQTT or another type of integration ?

### DISCUSSION THREAD

**Discussion by BIIINO** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#339] [General]
Hello,

is there any way how to control and monitor hot tubes by domoticz ?  Any guide or how to ?

Thank you very much in advance

**Comment by visualapproach** [comment]
If it supports MQTT yes. Or web sockets.
How to do it? No clue, but if you show the build instructions to someone who knows your home automation system I think you will get a better answer.

**Reply by BIIINO** [reply]
Thank you for your reply. Is there somewhere documented Weboscket please ?  More information about domoticz MQTT protocol are here:
https://www.domoticz.com/wiki/MQTT#Domoticz_MQTT_communication


Thank you in advance.

**Reply by visualapproach** [reply]
Not per se but it's pretty much the same as MQTT. Same commands and same JSON formatting. If you look at websockets.js you see how it can be done.
I don't have time to learn every home automation system out there. There is a protocol exposed to the world via mqtt and ws and it's up to the other side to use it. I'm happy to answer questions about the protocol on the ESP side.
If the users of system X want to add something to the project then they are welcome to tell me how it should be done.
But to give you a starting point: ESP send information as JSON strings, containing key/value pairs. You need to extract them and feed them to whatever sensor you've created. Usually there is built in functions for parsing JSON. In the case of MQTT the info is sent to several topics. For ws there is a key 'CONTENT' that tells what info is sent.
Wish you luck
